plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/config/OpenLineageClientHttpTransportConfig.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.openlineage.config;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class OpenLineageClientHttpTransportConfig
+{
+    private String url;
+    private String endpoint;
+    private Optional<String> apiKey = Optional.empty();
+    private Integer timeoutInMillis = 5000;
+    private Map<String, String> headers = new HashMap<>();
+    private Map<String, String> urlParams = new HashMap<>();
+
+    @NotNull
+    public String getUrl()
+    {
+        return url;
+    }
+
+    @ConfigDescription("URL of receiving server. Explicitly set the scheme https:// to use symmetric encryption")
+    @Config("openlineage-event-listener.openlineage.transport.url")
+    public OpenLineageClientHttpTransportConfig setUrl(String url)
+    {
+        this.url = url;
+        return this;
+    }
+
+    public String getEndpoint()
+    {
+        return endpoint;
+    }
+
+    @ConfigDescription("Custom path . Explicitly set the scheme https:// to use symmetric encryption")
+    @Config("openlineage-event-listener.openlineage.transport.endpoint")
+    public OpenLineageClientHttpTransportConfig setEndpoint(String endpoint)
+    {
+        this.endpoint = endpoint;
+        return this;
+    }
+
+    public Optional<String> getApiKey()
+    {
+        return apiKey;
+    }
+
+    @ConfigDescription("API Key to use when authenticating against OpenLineage API")
+    @Config("openlineage-event-listener.openlineage.transport.apiKey")
+    public OpenLineageClientHttpTransportConfig setApiKey(String apiKey)
+    {
+        this.apiKey = Optional.ofNullable(apiKey);
+        return this;
+    }
+
+    public Integer getTimeoutInMillis()
+    {
+        return timeoutInMillis;
+    }
+
+    @ConfigDescription("Timeout (milliseconds) when making HTTP Requests.")
+    @Config("openlineage-event-listener.openlineage.transport.timeoutInMillis")
+    public OpenLineageClientHttpTransportConfig setTimeoutInMillis(Integer timeoutInMillis)
+    {
+        this.timeoutInMillis = timeoutInMillis;
+        return this;
+    }
```

Comment:
My intention was to stick 1:1 to OL client data types so other OL integrations have same configuration approach - for example in spark config is also explicitly what client expects https://openlineage.io/docs/integrations/spark/configuration/transport

Reply:
But in case of Trino we tend to make all the configuration (across plugin or various module) to be similar way so any timeout specific configuration property would be generally injected as `Duration` from Airlift's unit library which would be unified across the project.

Reply:
added via https://github.com/trinodb/trino/pull/21265/commits/a36ed5a509146e10397e62b86769b6038f5e40fe